### ISSUE TITLE
Is this a typo?

### ISSUE THREAD

**Issue by 5hadowblad3** [sslab-gatech/qsym#16]
In solver.cpp of pintool:
`void Solver::negatePath(ExprRef e, bool taken) {
  reset();
  syncConstraints(e);
  addToSolver(e, !taken);
  bool sat = checkAndSave();
  if (!sat) {
    reset();
    // optimistic solving
    addToSolver(e, !taken);
    checkAndSave("optimistic");
  }
}`

Is the optimistic solving should have not negate "taken" as parameter rather than current one?

**Comment by jakkdu** [comment]
No. `taken` denotes that a current branch is taken by concrete execution or not.
QSYM wants to find an input that negates this branch to find a new path.
So it should be `!taken`.

The difference between normal and optimistic solving is whether to add another constraints or not (by `syncConstraints(e)`).
